scipy/sparse/base.py

```diff
@@ -1136,6 +1136,18 @@ def diagonal(self, k=0):
         """
         return self.tocsr().diagonal(k=k)
 
+    def trace(self, k=0):
```

Comment:
In numpy, this `k` parameter is named `offset`. We should use the same name here.

Reply:
I see the methods in base class `spmatrix`, both `diagonal` and `setdiag`  used this parameter (k=0), Could I change `k` to `offset` for all of them?

Reply:
If feasible, I'll submit the modification of `trace` firstly, and then I build other PR to submit the change of  `diagonal` and `setdiag`.

Reply:
Unfortunately the name of keyword arguments is effectively part of the public API, so changing existing methods isn't feasible.